ui/src/state/channel/channel.ts

```diff
@@ -583,13 +580,22 @@ export function useInfinitePosts(
     retry: false,
   });
 
+  // we stringify the data here so that we can use it in useMemo's dependency array.
```

Comment:
It doesn't, but that's just because we still invalidate on every update in useInfiniteDMs. If you comment out `invalidate.current()` you'll see the same behavior.